modules/library/jdbc/src/main/java/org/geotools/jdbc/JDBCDataStoreFactory.java

```diff
@@ -146,7 +149,20 @@
             "SQL statement executed when the connection is released to the pool", false, null,
             Collections.singletonMap(Parameter.IS_LARGE_TEXT, Boolean.TRUE));
 
-    
+    /** JDBC callback factory */
+    public static final Param CALLBACK_FACTORY = new Param("Callback factory", String.class, 
+        "Name of JDBCReaderCallbackFactory to enable on the data store", false);
+
+    /**
+     * system property  name used to set callback factory
+     */
+    static final String CALLBACK_PROPERTY = "gt2.jdbc.callback";
```

Comment:
@bencaradocdavies Yeah, I was going for consistency with the existing property "gt2.jdbc.trace".... but if folks like that style better I am happy to go that way.